### PR TITLE
Removed pylint directive and added clear configuration method

### PR DESCRIPTION
**Pull request by Akshat111111**

### CODE REVIEW

File: novu/config.py [hunk 1]
```diff
@@ -1,23 +1,46 @@
 """This module is used to define a configuration ``NovuConfig`` reusable through the package."""
 from novu.helpers import Singleton
 
+class NovuConfig(metaclass=Singleton):                                           # pylint: disable=R0903
```
Comment (ryshu) [inline novu/config.py:4]: Does not respect the indentation imposed by black to standardize the code

File: novu/config.py [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
-    """Singleton used to configure globally URL and API_KEY for the Novu Client"""
+    def __init__(self):
+        # Initialize default values or load them from a configuration file
+        self._url = ""
+        self._api_key = ""
 
-    url: str
-    """The URL of the Novu API to reach"""
+    @property
+    def url(self):
```
Comment (ryshu) [inline novu/config.py:13]: Is not documented, although the field was before

File: novu/config.py [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
 
-    api_key: str
-    """The API Key required to auth on the Novu API"""
+    @url.setter
+    def url(self, value):
+        """Set the URL of the Novu API to reach."""
+        self._url = value
+
+    @property
+    def api_key(self):
```
Comment (ryshu) [inline novu/config.py:22]: Is not documented, although the field was before

File: novu/config.py [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
 
     @classmethod
     def configure(cls, url: str, api_key: str):
-        """Class method provided to initialise the singleton data.
+        """Class method provided to configure the singleton data.
 
         Args:
-            url: The URL of the Novu API to reach.
-            api_key: The API Key required to auth on the Novu API.
+            url (str): The URL of the Novu API to reach.
```
Comment (ryshu) [inline novu/config.py:35]: Duplication of typing which is not useful

File: novu/config.py [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+            api_key (str): The API Key required to authenticate with the Novu API.
```
Comment (ryshu) [inline novu/config.py:36]: Duplication of typing which is not useful

File: novu/config.py [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
+    def api_key(self):
+        return self._api_key
+
+    @api_key.setter
+    def api_key(self, value):
+        """Set the API Key required to authenticate with the Novu API."""
+        self._api_key = value
 
     @classmethod
     def configure(cls, url: str, api_key: str):
```
Comment (ryshu) [inline novu/config.py:31]: The return is not documented

**Review by ryshu — CHANGES_REQUESTED** [review]
In personal opinion, this MR does not bring much to the bookstore but why not.

However, you must take the time to follow the contribution code. Here it is very easy for me to see that these have not been read:

* commit name is not correct
* code formatting is not applied, implying that pre-commit is not installed
* the branch is directed to main, where they must be directed to alpha
* the code is not tested

The description doesn't explain at all what this change does:

1. Here, we put getters/setters which serve no purpose (apart from adding code to maintain and reading complexity)
2. Add a `clear_configuration` method which sets the fields to "" values

Check with the depot managers but for me these changes are not useful at all and do not address any needs.

@Cliftonz @unicodeveloper WDYT ?

File: novu/config.py (hunk 6 above)
Comment (Akshat111111) [reply novu/config.py:31]: Sure, I will definately take care of the above points and will change it accordingly

**Pull request closed without merging**